plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -74,6 +77,31 @@ data ScriptDecodeError
 
 makeClassyPrisms ''ScriptDecodeError
 
+instance Pretty ScriptDecodeError where
+  pretty = \case
+    CBORDeserialiseError e ->
+      "Failed to deserialise a script:" <+> pretty e
+    RemainderError bs ->
+      "Script was successfully deserialised, but"
+        <+> pretty (BSL.length bs)
+        <+> "more bytes were encountered after the script's position."
+    LedgerLanguageNotAvailableError{..} ->
+      "You are attempting to use a"
+        <+> pretty sdeAffectedLang
+        <+> "script which is not yet supported by the current major protocol version"
+        <+> pretty sdeThisPv <> "."
+        <+> "The major protocol version that introduces this language is"
+        <+> pretty sdeIntroPv <> "."
+        <+> "To learn more about Plutus Core language versions, see \
+        \https://plutus.readthedocs.io/en/latest/explanations/language-versions.html."
+    PlutusCoreLanguageNotAvailableError{..} ->
```

Comment:
While you are at it, can you add `PlutusLedgerLanguage` to `PlutusCoreLanguageNotAvailableError`?

A Plutus Core version (such as 1.1.0) is introduced in a specific `PlutusLedgerLanguage` (e.g., PlutusV3) and a specific protocol version (e.g., 9). So, a `PlutusCoreLanguageNotAvailableError` may occur because of either the protocol version, or the `PlutusLedgerLanguage`.